Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/PersonStats.hs

```diff
@@ -0,0 +1,160 @@
+
+data UserCategory = POWER | REGULAR | IRREGULAR | RARE
+  deriving (Generic, ToJSON, FromJSON, ToSchema)
+
+getPersonStats :: (EsqDBReplicaFlow m r, EncFlow m r, SCC.CacheFlow m r, EsqDBFlow m r, CoreMetrics m) => (Id DP.Person, Id Merchant.Merchant) -> m PersonStatsRes
+getPersonStats (personId, merchantId) = do
+  person <- runInReplica $ QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+  personStats_ <- runInReplica $ QPS.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+  now <- getCurrentTime
+  when (all (== 0) [personStats_.userCancelledRides, personStats_.completedRides, personStats_.weekendRides, personStats_.weekdayRides, personStats_.offPeakRides, personStats_.eveningPeakRides, personStats_.morningPeakRides, personStats_.weekendPeakRides]) $ do
+    personStatsValues <- SP.backfillPersonStats personId merchantId
+    QPS.incrementOrSetPersonStats personStatsValues
+  -- Esq.runNoTransaction $ QPS.incrementOrSetPersonStats personStatsValues
+
+  personStats <- runInReplica $ QPS.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+  latestCompletedRide <- runInReplica $ QRide.findLatestCompletedRide personId
+  completedRidesCount <- runInReplica $ QRide.countRidesByRiderId personId
+  favLocNum <- runInReplica $ QSRL.countAllByRiderId personId
+  completedRidesInLast30DaysCount <- runInReplica $ QRide.countRidesFromDateToNowByRiderId personId (negate days30 `addUTCTime` now)
+  latestSearch <- runInReplica $ QSR.findLatestSearchRequest personId
+  decEmail <- decrypt `mapM` person.email
+  emergencyContactsNum <- length <$> runInReplica (QPDEN.findAllByPersonId personId)
+  let userCancellationRate =
+        fromIntegral personStats.userCancelledRides
+          / fromIntegral (personStats.userCancelledRides + personStats.completedRides)
+      overalCancellationRate =
+        fromIntegral (personStats.userCancelledRides + personStats.driverCancelledRides)
+          / fromIntegral (personStats.userCancelledRides + personStats.driverCancelledRides + personStats.completedRides)
+      weekdayMorningPeakRidesRate =
+        fromIntegral personStats.morningPeakRides
+          / fromIntegral personStats.completedRides
+      weekdayEveningPeakRidesRate =
+        fromIntegral personStats.eveningPeakRides
+          / fromIntegral personStats.completedRides
+      offPeakRidesRate =
+        fromIntegral personStats.offPeakRides
+          / fromIntegral personStats.completedRides
+      weekendRidesRate =
+        fromIntegral personStats.weekendRides
+          / fromIntegral personStats.completedRides
+      weekdayRidesRate =
+        fromIntegral personStats.weekdayRides
+          / fromIntegral personStats.completedRides
+      weekendPeakRideRate =
+        fromIntegral personStats.weekendPeakRides
+          / fromIntegral personStats.completedRides
+      daysFromRegDate :: Int = roundToIntegral $ diffUTCTime now person.createdAt / 60 / 60 / 24
```

Comment:
Any specific reason to do it this way - `/ 60 / 60 / 24` ?

Reply:
not really, Alexander worked on this task, so some changes are not mine .. this is one of them